src/llmq/quorums_signing_shares.cpp

```diff
@@ -1356,7 +1357,10 @@ void CSigSharesManager::WorkThreadMain()
         bool didWork = false;
 
         RemoveBannedNodeStates();
+        // TODO this got clogged with calls not belonging to sig shares handling, so we should consider introducing
+        // a dedicated place for coordination of these calls
         didWork |= quorumSigningManager->ProcessPendingRecoveredSigs(*g_connman);
+        didWork |= quorumInstantSendManager->ProcessPendingInstantSendLocks();
```

Comment:
Why not start InstantSend dedicated thread in CInstantSendManager instead?

Reply:
Hmm good question, didn't think about that. I'll look into doing this :)

Reply:
Did this now and force-pushed